### PR TITLE
Added minor improvements

### PR DESCRIPTION
**Pull request by rskrzypczak**
-Removed unused variable
-Added ability to open record
image

### CODE REVIEW

File: modules/Vtiger/models/Record.php [hunk 1]
```diff
@@ -1173,14 +1223,10 @@ public function initInventoryData()
...
-		$privilegesName = ['isEditable', 'isCreateable', 'isViewable', 'isNoLockByField'];
-		foreach ($privilegesName as $name) {
-			if (!empty($name) && isset($this->privileges[$name])) {
-				unset($this->privileges[$name]);
-			}
-		}
+		$this->privileges = [];
 		Users_Privileges_Model::clearLockEditCache($this->getModuleName() . $this->getId());
 		\vtlib\Functions::clearCacheMetaDataRecord($this->getId());
+		\App\Cache::delete('UnlockFields', $this->getId());
```
Comment (TomaszQr) [inline modules/Vtiger/models/Record.php]: staticDelete ?

File: modules/Vtiger/models/Record.php [hunk 2]
```diff
@@ -458,6 +458,7 @@ public function save()
 			$this->isNew = false;
 		}
 		\App\Cache::delete('recordLabel', $this->getId());
+		\App\Cache::delete('UnlockFields', $this->getId());
```
Comment (TomaszQr) [inline modules/Vtiger/models/Record.php]: staticDelete

File: layouts/basic/modules/Vtiger/Modals/Unlock.tpl [hunk 3]
```diff
@@ -0,0 +1,34 @@
+{*<!-- {[The file is published on the basis of YetiForce Public License 3.0 that can be found in the following directory: licenses/LicenseEN.txt or yetiforce.com]} -->*}
+{strip}
+	<form name="recordState" class="tpl-Base-Modals-RecordState" method="post">
```
Comment (mariuszkrzaczkowski) [inline layouts/basic/modules/Vtiger/Modals/Unlock.tpl]: layouts/basic/modules/Vtiger/Modals/Unlock.tpl >>
layouts/basic/modules/Vtiger/Modals/RecordUnlock.tpl

File: modules/Vtiger/actions/Unlock.php [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+	}
+
+	/**
+	 * {@inheritdoc}
+	 */
+	public function process(\App\Request $request)
+	{
+
+		$recordModel = Vtiger_Record_Model::getInstanceById($request->getInteger('record'));
+		foreach ($recordModel->getUnlockFields() as $fieldName => $values) {
```
Comment (mariuszkrzaczkowski) [inline modules/Vtiger/actions/Unlock.php]: $fields = $recordModel->getUnlockFields() 
foreach ($fields as $fieldName => $values) {

File: modules/Vtiger/actions/Unlock.php [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+			if ($request->has($fieldName)) {
+				$recordModel->getField($fieldName)->getUITypeModel()->setValueFromRequest($request, $recordModel);
+			}
+		}
+		$recordModel->clearPrivilegesCache();
+		$result = false;
+		if (!$recordModel->getUnlockFields()) {
```
Comment (mariuszkrzaczkowski) [inline modules/Vtiger/actions/Unlock.php]: if (!$fields) {

File: public_html/layouts/basic/modules/Vtiger/resources/Unlock.js [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+	 * @param {jQuery} modalContainer
+	 */
+	registerEvents(modalContainer) {
+		modalContainer.find('form').on('submit', function (e) {
+			e.preventDefault();
+			const progressIndicator = $.progressIndicator({position: 'html', blockInfo: {enabled: true}});
+			AppConnector.request($(this).serializeFormData()).done((response) => {
+				let result = response.result;
+				if (result.success && result.url) {
+					window.location.href = result.url;
```
Comment (mariuszkrzaczkowski) [inline public_html/layouts/basic/modules/Vtiger/resources/Unlock.js]: app.openUrl

**Pull request merged**